sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -349,11 +349,12 @@ object MapObjects {
   def apply(
       function: Expression => Expression,
       inputData: Expression,
-      elementType: DataType): MapObjects = {
+      elementType: DataType,
+      inputDataType: Option[DataType] = None): MapObjects = {
```

Comment:
At most scenarios, we don't need to specify inputDataType. So I think it is indeed the obvious value. I agreed that the use of option here is not good. I will change it and have it documented.